src/_helpers/auth.ts

```diff
@@ -0,0 +1,92 @@
+import jwt from 'jsonwebtoken';
+import { logger } from "../index";
+import { currentConfig } from './config';
+import { clone } from "./clone";
+
+import { AuthenticateSuccessResponse } from '../_models/AuthenticateSuccessResponse';
+import { User } from '../_models/User';
+
+export function authenticate(username: string, password: string): Promise<AuthenticateSuccessResponse> {
+    return new Promise<AuthenticateSuccessResponse>((resolve, reject) => {
+        console.log(currentConfig.users);
+        let userFound = false;
+        currentConfig.users.forEach((user_original) => {
+            let user = clone(user_original);
+            if(username === user.username) {
+                userFound = true;
+                if(password !== user.password) {
+                    reject(new Error('Password is incorrect'));
+                }
+                delete user["password"];
+                jwt.sign({user}, currentConfig.security.jwt_private_key, { expiresIn: '2 days' }, (err, token) => {
+                    if(err) { reject(err) }
+                    resolve({
+                        access_token: token,
+                        user: user
+                    });
+                });
+                return true;
+            }
+        });
+        if(!userFound) reject(new Error('User not found'));
+    });
+}
+
+export function validateAccessToken(access_token: string): Promise<User> {
+    return new Promise<User>((resolve, reject) => {
+        jwt.verify(access_token, currentConfig.security.jwt_private_key, (err, decoded) => {
```

Comment:
I just tried it in GitPod and had the following problem:
I first checked out the `master` branch and ran TA. It created the config file with the defaults because it did not exist. Then, I checked out the `improve-authentication` branch and got a bunch of errors, because `currentConfig.security.jwt_private_key` is `undefined`... 

Do users have to create a new config file for `v3.0.0` anyway or should we build some kind of migration which adds this to the config?